### PR TITLE
H5 tests

### PR DESCRIPTION
**Pull request by rafia17**
I have changed two files:

h5.py: added the checks to make sure that the length of y and dataset arrays matches the timeseries array. Raise a ValueError if it doesn't

test_h5.py: added tests to check if the ValueError is raised appropriately

### CODE REVIEW

File: libs/io/bbhnet/io/h5.py
Comment (alecgunny) [inline libs/io/bbhnet/io/h5.py:8]: This should be under the TYPE_CHECKING `if` statement, correct?

File: libs/io/bbhnet/io/h5.py
Comment (alecgunny) [inline libs/io/bbhnet/io/h5.py:42]: Consider using the `key` in this error so its more explicit as to which array is the offender

File: libs/io/tests/test_h5.py
Comment (alecgunny) [inline libs/io/tests/test_h5.py:168]: Small aesthetic point, but since `datasets[key]` is just an array, you actually don't need the `[:]` slice, which is just used as syntatic sugar for HDF5 `dataset` objects to extract the underlying data. It will still _work_ for arrays, but it's just saying "return a full slice of the array", which it would have done anyway.

**Review by alecgunny — CHANGES_REQUESTED** [review]
The tests look good, just a couple minor points of feedback. It also looks like you currently have a merge conflict with the main branch, so you'll need to fix that before the CI tests can run.

File: libs/io/bbhnet/io/h5.py
Comment (rafia17) [reply libs/io/bbhnet/io/h5.py]: Done

File: libs/io/bbhnet/io/h5.py
Comment (rafia17) [reply libs/io/bbhnet/io/h5.py]: Thanks for explaining. I have made the change

File: libs/io/bbhnet/io/h5.py
Comment (rafia17) [reply libs/io/bbhnet/io/h5.py]: Added the "key" to the error message

File: libs/io/tests/test_h5.py
Comment (rafia17) [reply libs/io/tests/test_h5.py]: removed the [:]

**Comment by rafia17** [comment]
Hi Alec, I have incorporated the minor changes that you had requested and it needs your approval to merge.

File: libs/io/bbhnet/io/h5.py
Comment (alecgunny) [inline libs/io/bbhnet/io/h5.py:46]: this will be rendered better as a single string: `f"Length of data array '{key}' doesn't match the length of timeseries"`

**Review by alecgunny — CHANGES_REQUESTED** [review]
Looking great! One last small change and it's ready to merge (assuming the tests pass)

File: libs/io/bbhnet/io/h5.py
Comment (rafia17) [reply libs/io/bbhnet/io/h5.py]: This change is incorporated

**Review by alecgunny — APPROVED** [review]

**Pull request merged**
